control/plugin/client/grpc.go

```diff
@@ -241,12 +281,142 @@ func (g *grpcClient) CollectMetrics(mts []core.Metric) ([]core.Metric, error) {
 	return metrics, nil
 }
 
+func (g *grpcClient) UpdateCollectedMetrics(mts []core.Metric) error {
+	if g.stream != nil {
+		arg := &rpc.CollectArg{
+			Metrics_Arg: &rpc.MetricsArg{Metrics: NewMetrics(mts)},
+		}
+		err := g.stream.Send(arg)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+func (g *grpcClient) UpdatePluginConfig(bytes []byte) error {
+	if g.stream != nil {
+		arg := &rpc.CollectArg{
+			Other: bytes,
+		}
+		err := g.stream.Send(arg)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+func (g *grpcClient) UpdateCollectDuration(maxCollectDuration time.Duration) error {
+	if g.stream != nil {
+		arg := &rpc.CollectArg{
+			MaxCollectDuration: maxCollectDuration.Nanoseconds(),
+		}
+		err := g.stream.Send(arg)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+func (g *grpcClient) UpdateMetricsBuffer(maxMetricsBuffer int64) error {
+	if g.stream != nil {
+		arg := &rpc.CollectArg{
+			MaxMetricsBuffer: maxMetricsBuffer,
+		}
+		err := g.stream.Send(arg)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (g *grpcClient) StreamMetrics(mts []core.Metric) (chan []core.Metric, chan error, error) {
```

Comment:
I'm not sure I understand what you mean. Could you give an example so I can wrap my head around it?